src/rttable.c

```diff
@@ -494,10 +548,16 @@ void setRouteLastMemberMode(uint32_t group) {
         // Check for fast leave mode...
         if(croute->upstrState == ROUTESTATE_JOINED && conf->fastUpstreamLeave) {
             // Send a leave message right away only when the route has been active on only one interface
-            if (numberOfInterfaces(croute) <= 1) {
-                my_log(LOG_DEBUG, 0, "Leaving group %d now", group);
+            if (countDownstreamHosts(croute) <= 1) {
+                my_log(LOG_DEBUG, 0, "Fastleave is enabled and this was the last downstream host, leaving group %s now", inetFmt(croute->group, s1));
                 sendJoinLeaveUpstream(croute, 0);
-            }
+            } else {
+                int hostIdx = findDownstreamSlot(croute, src);
+                if(hostIdx != -1){
+                    croute->dwnstrHosts[hostIdx] = 0;
+                }
+            my_log(LOG_DEBUG, 0, "Fast leave enabled but there are still %d downstream hosts left, not leaving group %s", countDownstreamHosts(croute), inetFmt(croute->group, s1));
+        }
         }
```

Comment:
This looks like broken indentation.

Reply:
Fixed that in a later commit, I can squash all related commits into one if you prefer that. No problem.

Reply:
Yes, squashing changes which implements or fixes one thing to one commit is a good idea.